### PR TITLE
Add tests to system culling, update opt_out procedure

### PR DESCRIPTION
**Pull request by semtexzv**

### CODE REVIEW

File: database_admin/migrations/018_caches_fix.up.sql [hunk 1]
```diff
@@ -0,0 +1,114 @@
+CREATE OR REPLACE FUNCTION mark_stale_systems()
```
Comment (Josca) [inline database_admin/migrations/018_caches_fix.up.sql:1]: Shouldn't we add previous method version into down.sql script?

File: database_admin/schema/create_schema.sql [hunk 2]
```diff
@@ -159,6 +156,7 @@ BEGIN
     FROM system_advisories sa
     WHERE sa.system_id = NEW.id
       AND sa.when_patched IS NULL
+      AND change > 0
```
Comment (Josca) [inline database_admin/schema/create_schema.sql:159]: Just to be sure, isn't it be `change != 0`?

File: database_admin/migrations/018_caches_fix.up.sql (hunk 1 above)
Comment (semtexzv) [reply database_admin/migrations/018_caches_fix.up.sql:1]: Previous version had a significant mistake, I don't see a reason to have it around.

File: database_admin/schema/create_schema.sql (hunk 2 above)
Comment (semtexzv) [reply database_admin/schema/create_schema.sql:159]: No, this is a logical error. It should never happen, but in the old code we performed insertion only when incrementing the counters.

**Pull request merged**